engine/lite/k8sDeploymentManager/src/main/scala/pl/touk/nussknacker/k8s/manager/K8sDeploymentManagerProvider.scala

```diff
@@ -0,0 +1,62 @@
+package pl.touk.nussknacker.k8s.manager
+
+import akka.actor.ActorSystem
+import com.typesafe.config.Config
+import pl.touk.nussknacker.engine.ModelData.BaseModelDataExt
+import pl.touk.nussknacker.engine.api.component.AdditionalPropertyConfig
+import pl.touk.nussknacker.engine.api.deployment.{DeploymentManager, ProcessingTypeDeploymentService}
+import pl.touk.nussknacker.engine.api.process.ProcessName
+import pl.touk.nussknacker.engine.api.{LiteStreamMetaData, RequestResponseMetaData}
+import pl.touk.nussknacker.engine.requestresponse.api.openapi.RequestResponseOpenApiSettings
+import pl.touk.nussknacker.engine.{BaseModelData, CustomProcessValidator, DeploymentManagerProvider, TypeSpecificInitialData}
+import pl.touk.nussknacker.k8s.manager.RequestResponseSlugUtils.defaultSlug
+import sttp.client.{NothingT, SttpBackend}
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/*
+  Each scenario is deployed as Deployment+ConfigMap
+  ConfigMap contains: model config with overrides for execution and scenario
+ */
+class K8sDeploymentManagerProvider extends DeploymentManagerProvider {
+
+  override def createDeploymentManager(modelData: BaseModelData, config: Config)
+                                      (implicit ec: ExecutionContext, actorSystem: ActorSystem,
+                                       sttpBackend: SttpBackend[Future, Nothing, NothingT],
+                                       deploymentService: ProcessingTypeDeploymentService): DeploymentManager = {
+    new K8sDeploymentManager(modelData.asInvokableModelData, K8sDeploymentManagerConfig.parse(config))
+  }
+
+  private val steamingInitialMetData = TypeSpecificInitialData(LiteStreamMetaData(Some(1)))
+
+  override def typeSpecificInitialData(config: Config): TypeSpecificInitialData = {
+    forMode(config)(
+      _ => steamingInitialMetData,
+      config => (scenarioName: ProcessName, _: String) => RequestResponseMetaData(Some(defaultSlug(scenarioName,
+        config.nussknackerInstanceName)))
+    )
+  }
+
+  override def additionalPropertiesConfig(config: Config): Map[String, AdditionalPropertyConfig] = forMode(config)(
+    _ => Map.empty,
+    _ => RequestResponseOpenApiSettings.additionalPropertiesConfig
+  )
+
+  override def additionalValidators(config: Config): List[CustomProcessValidator] = forMode(config)(
+    _ => Nil,
+    config => List(new LiteScenarioValidator(config.nussknackerInstanceName))
```

Comment:
What about validatePreparedService? IMO  we should run both validations:
- slug acceptable characters and length restrictions
- slug uniqueness
at once dynamically during validation of scenario.

Reply:
I would rather validate uniqueness only during deploy. We need access to K8s then, and it's more along the lines "transient errors (like duplication)" are done during deploy...

Reply:
On the other hand it is very easy to duplicate slug - let's say someone do export -> import of scenario. We validate scenario name, things with slug looks the same for me. Maybe we could do validation only on designer side by some query on DB :thinking: or some short living cache for this info from k8s. WDYT?

Reply:
We could... but I'd leave it for separate PR not to block the API

Reply:
ok